Failed to create EpollDatagramChannel after upgrading to 4.2.8.Final

After upgrading to 4.2.8.Final, this exception started to raise:
```
Caused by: java.lang.reflect.InvocationTargetException: null
    at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
    at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
    at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
    at io.netty.channel.ReflectiveChannelFactory.newChannel(ReflectiveChannelFactory.java:44)
    ... 110 common frames omitted
Caused by: io.netty.channel.ChannelException: setsockopt() failed: Protocol not available
    at io.netty.channel.epoll.LinuxSocket.setIpMulticastAll(Native Method)
    at io.netty.channel.epoll.LinuxSocket.setIpMulticastAll(LinuxSocket.java:213)
    at io.netty.channel.epoll.EpollDatagramChannel.<init>(EpollDatagramChannel.java:132)
    at io.netty.channel.epoll.EpollDatagramChannel.<init>(EpollDatagramChannel.java:116)
    at io.netty.channel.epoll.EpollDatagramChannel.<init>(EpollDatagramChannel.java:97)
    ... 116 common frames omitted
```

I think this catch block must catch the `ChannelException`:
https://github.com/netty/netty/blob/4.2/transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java#L133

This is the OS information where the code is executed:
```
cat /etc/os-release

PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian

uname -a
Linux <hostname> 4.18.0-553.el8_10.x86_64 #1 SMP Fri May 24 13:05:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

--- a/transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java
+++ b/transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java
@@ -19,6 +19,7 @@ import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufAllocator;
 import io.netty.buffer.Unpooled;
 import io.netty.channel.AddressedEnvelope;
+import io.netty.channel.ChannelException;
 import io.netty.channel.ChannelFuture;
 import io.netty.channel.ChannelMetadata;
 import io.netty.channel.ChannelOutboundBuffer;
@@ -130,7 +131,7 @@ public final class EpollDatagramChannel extends AbstractEpollChannel implements 
         // Configure IP_MULTICAST_ALL - disable by default to match the behaviour of NIO.
         try {
             fd.setIpMulticastAll(IP_MULTICAST_ALL);
-        } catch (IOException e) {
+        } catch (IOException | ChannelException e) {
             logger.debug("Failed to set IP_MULTICAST_ALL to {}", IP_MULTICAST_ALL, e);
         }
 

--- a/transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringDatagramChannel.java
+++ b/transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringDatagramChannel.java
@@ -17,6 +17,7 @@ package io.netty.channel.uring;
 
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.AddressedEnvelope;
+import io.netty.channel.ChannelException;
 import io.netty.channel.ChannelFuture;
 import io.netty.channel.ChannelMetadata;
 import io.netty.channel.ChannelOutboundBuffer;
@@ -119,7 +120,7 @@ public final class IoUringDatagramChannel extends AbstractIoUringChannel impleme
         // Configure IP_MULTICAST_ALL - disable by default to match the behaviour of NIO.
         try {
             fd.setIpMulticastAll(IP_MULTICAST_ALL);
-        } catch (IOException e) {
+        } catch (IOException | ChannelException e) {
             logger.debug("Failed to set IP_MULTICAST_ALL to {}", IP_MULTICAST_ALL, e);
         }
 

